libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -61,6 +61,60 @@ typedef struct _packetMatchParams
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Handle special corner cases regarding terminal wildcards, as
```

Comment:
Terminal meaning at the end of the string?

Is there a specific section of (a specific version of) the MQTT standard that we could reference? I have found these feature/standard references to be tough to audit after the fact.

Reply:
There is: 
http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/csprd02/mqtt-v3.1.1-csprd02.html#_Toc385349843

The specific requirements this function checks are:
>  “sport/#” also matches the singular “sport”, since # includes the parent level. 
and
> “sport/+” does not match “sport” but it does match “sport/”